### PR TITLE
feat(ingestion) Allow for ingestion to read files remotely

### PR DESCRIPTION
**Pull request by xiphl**
The proposition of this PR is to allow the following sources to read from URLs (especially git repos) instead of being constrained to local files:
1. csv-enricher
2. business glossary
3. file
4. file-lineage

Still work in progress. (Especially updating the tests)

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by shirshanka** [comment]
Excited for this! Support for s3, gcs would be great to add as well.

**Comment by xiphl** [comment]
I will try to finish the missing bits soon!
I'm not sure how to implement for S3/GCS sources though.

**Comment by xiphl** [comment]
let me work on the failed tests first

**Comment by xiphl** [comment]
@shirshanka I could do with some advice on the failed assert in auditStamp time between the golden and generated json file.  
For some reason the time set by freezegun isn't picked up when its at the file-lineage logic, yet I don't see any reason why that would be the case when that module had similar approaches to generating the auditStamp. Reordering the tests doesnt change the fact that it always fail at file-lineage.
Any ideas?

File: metadata-ingestion/tests/integration/remote/test_remote.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+from datahub.ingestion.run.pipeline import Pipeline
+from tests.test_helpers import mce_helpers
+from tests.test_helpers.docker_helpers import wait_for_port
+
+FROZEN_TIME = "2021-08-24 09:00:00"
+
+
+@freeze_time(FROZEN_TIME)
+@pytest.mark.integration
+def test_remote_ingest(docker_compose_runner, pytestconfig, tmp_path, mock_time):
```
Comment (asikowitz) [inline metadata-ingestion/tests/integration/remote/test_remote.py:13]: Hey @xiphl, the `mock_time` fixture you're using here also mocks `time.time()` which is what generates audit stamps. I think this test will pass if you remove the fixture!

Comment (xiphl) [reply metadata-ingestion/tests/integration/remote/test_remote.py:13]: yup that was the cause indeed!

File: metadata-ingestion/src/datahub/ingestion/source/metadata/lineage.py [hunk 2]
```diff
@@ -160,7 +155,6 @@ def _get_entity_urn(entity_config: EntityConfig) -> Optional[str]:
                             new_upstream = models.UpstreamClass(
                                 dataset=upstream_entity_urn,
                                 type=models.DatasetLineageTypeClass.TRANSFORMED,
-                                auditStamp=auditStamp,
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/lineage.py:163]: Can you keep this, but use an auditStamp created in this function?

**Review by asikowitz — APPROVED** [review]
This looks great! Thanks so much for working on this and contributing further to datahub. I left a few docs / style comments, mostly for my own benefit; I think this is good to go.

**Pull request merged**